integ-tests/src/main/java/com/google/testdata/utils/PerformanceTestDataGeneratorTask.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.testdata.utils;
+
+import static com.google.sql.SqlHelper.connectAndExecuteQueryAsUser;
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+
+import com.google.common.base.Joiner;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.RecursiveAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Executes queries for each user in separate thread. */
+public class PerformanceTestDataGeneratorTask extends RecursiveAction {
```

Comment:
Why we can't place all this code into `PerformanceTestDataHelper` as private methods?
They are solely used by one class - `PerformanceTestDataGenerator` and could be not exposed.

Reply:
This entity is executed by ForkJoin api, so should be separate

Reply:
Ok, makes sense, so, if it's an executor maybe we want to reflect it somehow in the name of the class? Otherwise, what does 'Task' indicate for any one who will be working with the code?

Reply:
Done